### PR TITLE
Restore index versions 7 in lucene_snapshot_10

### PR DESCRIPTION
**Pull request by javanna**

### CODE REVIEW

**Review by ChrisHegarty — APPROVED** [review]
LGTM

File: modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java [hunk 1]
```diff
@@ -390,7 +391,7 @@ public void testParse3DPolygon() throws IOException, ParseException {
 
         LinearRing shell = GEOMETRY_FACTORY.createLinearRing(shellCoordinates.toArray(new Coordinate[shellCoordinates.size()]));
         Polygon expected = GEOMETRY_FACTORY.createPolygon(shell, null);
-        final IndexVersion version = IndexVersionUtils.randomCompatibleVersion(random());
+        final IndexVersion version = IndexVersionUtils.randomPreviousCompatibleVersion(random(), IndexVersions.V_8_0_0);
```
Comment (cbuescher) [inline modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java:394]: I'm curious why this doesn't fail, MININUM_COMPATIBLE should be at 8.0.0 but we should work with 7.9.0 here. Just a remark though

File: server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java [hunk 2]
```diff
@@ -6638,8 +6638,6 @@ void assertLuceneOperations(InternalEngine engine, long expectedAppends, long ex
         assertThat(message, engine.getNumDocDeletes(), equalTo(expectedDeletes));
     }
 
-    // TODO Lucene 10 upgrade, we need to remove old IndexVersions for this to work
-    @AwaitsFix(bugUrl = "")
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java:6642]: Does this fail now?

**Review by cbuescher — COMMENTED** [review]
Just also went through all the changes to check, makes sense in terms of reverting but I wonder how some of the tests would pass now. Left two remarks around that.

File: modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java (hunk 1 above)
Comment (javanna) [reply modules/legacy-geo/src/test/java/org/elasticsearch/legacygeo/GeoJsonShapeParserTests.java:394]: I believe for all of these, I took what we have in main, that was adjusted with the 9x bump.

File: server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java (hunk 2 above)
Comment (javanna) [reply server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java:6642]: My main thinking is "with the 9x bump happened in main, why would we need to do anything special around index versions in lucene_snapshot_10?"

**Comment by cbuescher** [comment]
LGTM

**Pull request merged**